scripts/live-canary/test_notify_slack.py

```
@@ -435,6 +436,169 @@ def test_slack_payload_includes_pr_branch_context_when_present(self):
+                github_token="token",
+                run_context=notify.CanaryRunContext(target_pr="42abc"),
+                run_url=None,
+                commit=None,
+            )
+
+        with mock.patch.object(notify, "get_json", return_value={"number": 42}) as get_json:
+            with mock.patch.object(
+                notify,
+                "post_json",
+                return_value={"html_url": "https://github.com/nearai/ironclaw/pull/42#issuecomment-1"},
+            ) as post_json:
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)